apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts

```diff
@@ -26,15 +26,40 @@ export function useFaucetMutation(options?: UseFaucetMutationOptions) {
 				throw new Error('Failed, faucet host not found.');
 			}
 
-			const { error, transferredGasObjects } = await requestSuiFromFaucetV0({
+			const { error, task: taskId } = await requestSuiFromFaucetV1({
 				recipient: addressToTopUp,
 				host: options.host,
 			});
 
-			if (error) {
-				throw new Error(error);
+			if (error || !taskId) {
+				throw new Error(error ?? 'Failed, task id not found.');
 			}
-			return transferredGasObjects.reduce((total, { amount }) => total + amount, 0);
+
+			// Continuously check the status until it's no longer 'INPROGRESS'
+			let currentStatus = 'INPROGRESS';
+			while (currentStatus === 'INPROGRESS') {
```

Comment:
I think we should time-limit this so that we don't infinitely query in the event of downtime. Maybe just add a iteration count and limit it to some amount of iterations

Reply:
Limiting the request to a max 10,